File: entities/scanner/src/main/java/com/inquallity/daggersandbox/scanner/Scanner.java
package com.inquallity.daggersandbox.scanner;

import android.util.Log;

import javax.inject.Inject;

public class Scanner implements IScanner {

    @Inject
    public Scanner() {
    }

    @Override
    public void doScan() {
        Log.d("OYAEBU", "scanner doing scan $this");

    }

    @Override
    public void initialize() {
        Log.d("OYAEBU", "Do initialization of $this");
    }
}

File: entities/scanner/src/main/java/com/inquallity/daggersandbox/scanner/Scanner.kt
package com.inquallity.daggersandbox.scanner

import android.util.Log
import javax.inject.Inject

class Scanner @Inject constructor() : IScanner {

    override fun doScan() {
        Log.d("OYAEBU", "scanner doing scan $this")
    }

    override fun initialize() {
        Log.d("OYAEBU", "Do initialization of $this")
    }
}

interface IScanner {
    fun doScan()
    fun initialize()
}
